pkg/cache/clusterqueue_test.go

```diff
@@ -620,3 +621,143 @@ func TestClusterQueueUpdateWithAdmissionCheck(t *testing.T) {
 		})
 	}
 }
+
+func TestDominantResourceShare(t *testing.T) {
+	cases := map[string]struct {
+		cq          ClusterQueue
+		workload    *workload.Info
+		wantDRValue int
+		wantDRName  corev1.ResourceName
+	}{
+		"no cohort": {
+			cq: ClusterQueue{
+				ResourceStats: ResourceStats{
+					corev1.ResourceCPU: {
+						Nominal:  2_000,
+						Lendable: 2_000,
+						Usage:    1_000,
+					},
+					"example.com/gpu": {
+						Nominal:  5,
+						Lendable: 5,
+						Usage:    2_000,
+					},
+				},
+			},
+		},
+		"usage below nominal": {
+			cq: ClusterQueue{
+				ResourceStats: ResourceStats{
+					corev1.ResourceCPU: {
+						Nominal:  2_000,
+						Lendable: 2_000,
+						Usage:    1_000,
+					},
+					"example.com/gpu": {
+						Nominal:  5,
+						Lendable: 5,
+						Usage:    2,
+					},
+				},
+				Cohort: &Cohort{
+					ResourceStats: ResourceStats{
+						corev1.ResourceCPU: {
+							Nominal:  10_000,
+							Lendable: 10_000,
+							Usage:    2_000,
+						},
+						"example.com/gpu": {
+							Nominal:  10,
+							Lendable: 10,
+							Usage:    6,
+						},
+					},
+				},
+			},
+			wantDRName: corev1.ResourceCPU, // due to alphabetical order.
+		},
+		"usage above nominal": {
```

Comment:
I think it would be also worth adding a mixed test case - where we are below in one resource, but above in another to show that the max is returned.

Reply:
Done